sw/device/lib/testing/meson.build

```diff
@@ -0,0 +1,41 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# TODO: gtest should be vendored in.
+sw_lib_testing_gtest = declare_dependency(
+  include_directories: include_directories(
+    '/tools/googletest/googletest/include',
+    '/tools/googletest/googlemock/include',
+  ),
+  link_args: [
+    '-L/tools/googletest/lib',
+    '-lgtest',
+    '-lgtest_main',
+    '-lgmock',
+    '-lpthread',
+  ],
+)
+
+sw_lib_testing_mock_mmio = declare_dependency(
+  link_with: static_library(
+    'mock_mmio',
+    sources: ['mock_mmio.cc'],
+    dependencies: [sw_lib_testing_gtest],
+    native: true,
+    cpp_args: ['-DMOCK_MMIO'],
+  )
+)
+
+# Example test using mock_mmio.h, which also serves to
+# test that mock_mmio.h works correctly.
+test('mock_mmio_test', executable(
+  'mock_mmio_test',
+  sources: ['mock_mmio_test.cc'],
```

Comment:
I would expect to see `mmio.c` here. I guess this works because `mock_mmio_test.cc` does not use anything other than `reg32_read()` and `reg32_write()`.

Reply:
Technically, no. Since `mmio.h` is only used from a `.cc` file, the `extern` definitions in `mmio.c` are not necessary (C++ does not require that `inline` functions define a storage location, much like it doesn't for template functions... which are a type of inline function).

I may add it for consistency, but it's ultimately not necessary.